src/common/lib/client/pushchannel.ts

```diff
@@ -0,0 +1,117 @@
+import RestChannel from './restchannel';
+import { LocalDevice } from 'plugins/push/pushactivation';
+
+class PushChannel {
+  rest: BaseRest;
+  channel: RestChannel;
+
+  constructor(channel: RestChannel) {
+    this.channel = channel;
+    this.rest = channel.client;
+  }
+
+  async subscribeDevice() {
+    const rest = this.rest;
+    const device = rest.device as LocalDevice;
+    const format = rest.options.useBinaryProtocol ? rest.Utils.Format.msgpack : rest.Utils.Format.json,
+      body = { deviceId: device.id, channel: this.channel.name },
+      headers = rest.Defaults.defaultPostHeaders(rest.options, { format });
+
+    if (rest.options.headers) rest.Utils.mixin(headers, rest.options.headers);
+
+    rest.Utils.mixin(headers, this._getPushAuthHeaders());
+
+    const requestBody = rest.Utils.encodeBody(body, rest._MsgPack, format);
+    await rest.Resource?.post(rest, '/push/channelSubscriptions', requestBody, headers, {}, format, true);
+  }
+
+  async unsubscribeDevice() {
+    const rest = this.rest;
+    const device = rest.device as LocalDevice;
+    const format = rest.options.useBinaryProtocol ? rest.Utils.Format.msgpack : rest.Utils.Format.json,
+      headers = rest.Defaults.defaultPostHeaders(rest.options, { format });
+
+    if (rest.options.headers) rest.Utils.mixin(headers, rest.options.headers);
+
+    rest.Utils.mixin(headers, this._getPushAuthHeaders());
+
+    await rest.Resource?.delete(
+      rest,
+      '/push/channelSubscriptions',
+      headers,
+      { deviceId: device.id, channel: this.channel.name },
+      format,
+      true,
+    );
+  }
+
+  async subscribeClient() {
+    const rest = this.rest;
+    const clientId = this._getClientId();
+    const format = rest.options.useBinaryProtocol ? rest.Utils.Format.msgpack : rest.Utils.Format.json,
+      body = { clientId: clientId, channel: this.channel.name },
+      headers = rest.Defaults.defaultPostHeaders(rest.options, { format });
+
+    if (rest.options.headers) rest.Utils.mixin(headers, rest.options.headers);
+
+    const requestBody = rest.Utils.encodeBody(body, rest._MsgPack, format);
+    await rest.Resource?.post(rest, '/push/channelSubscriptions', requestBody, headers, {}, format, true);
+  }
+
+  async unsubscribeClient() {
+    const rest = this.rest;
+
+    const clientId = this._getClientId();
+    const format = rest.options.useBinaryProtocol ? rest.Utils.Format.msgpack : rest.Utils.Format.json,
+      headers = rest.Defaults.defaultPostHeaders(rest.options, { format });
+
+    if (rest.options.headers) rest.Utils.mixin(headers, rest.options.headers);
+
+    await rest.Resource?.delete(
+      rest,
+      '/push/channelSubscriptions',
+      headers,
+      { clientId: clientId, channel: this.channel.name },
+      format,
+      true,
+    );
+  }
+
+  async listSubscriptions(params?: Record<string, string>) {
+    this.rest.Logger.logAction(
+      this.rest.logger,
+      this.rest.Logger.LOG_MICRO,
```

Comment:
Yeah I thought about this and I don't think there's a worthwhile way to achieve log stripping for the plugin since there's no way to tell at import time whether the plugin is being used in a modular client or a default client. The only alternative I can think of is to offer a separate build of the plugin without logs but the size difference would be minimal so I think we should leave it as-is.